tests/director/tests/cert_subject_mapping_test.go

```diff
@@ -0,0 +1,187 @@
+	csmCreate := fixtures.CreateCertificateSubjectMapping(t, ctx, certSecuredGraphQLClient, csmInput)
+	defer fixtures.CleanupCertificateSubjectMapping(t, ctx, certSecuredGraphQLClient, csmCreate.ID)
+
+	queryCertSubjectMappingReq := fixtures.FixQueryCertificateSubjectMappingRequest(csmCreate.ID)
+	saveExample(t, queryCertSubjectMappingReq.Query(), "query certificate subject mapping")
+	csm := graphql.CertificateSubjectMapping{}
+
+	t.Logf("Query certificate subject mapping by ID: %s", csmCreate.ID)
+	err := testctx.Tc.RunOperationWithoutTenant(ctx, certSecuredGraphQLClient, queryCertSubjectMappingReq, &csm)
+	require.NoError(t, err)
+	require.NotEmpty(t, csm.ID)
+
+	assertions.AssertCertificateSubjectMapping(t, &csmInput, &csm)
+}
+
+func TestQueryCertSubjectMappings(t *testing.T) {
+	first := 100
+	queryCertSubjectMappingsWithPaginationReq := fixtures.FixQueryCertificateSubjectMappingsRequestWithPageSize(first)
+	saveExample(t, queryCertSubjectMappingsWithPaginationReq.Query(), "query certificate subject mappings")
+	currentCertSubjectMappings := graphql.CertificateSubjectMappingPage{}
+
+	t.Log("Getting current certificate subject mappings...")
+	err := testctx.Tc.RunOperationWithoutTenant(ctx, certSecuredGraphQLClient, queryCertSubjectMappingsWithPaginationReq, &currentCertSubjectMappings)
+	require.NoError(t, err)
+	t.Logf("Current number of certificate subject mappings is: %d", currentCertSubjectMappings.TotalCount)
+
+	// Create first certificate subject mapping
+	csmInput := fixtures.FixCertificateSubjectMappingInput(subject, consumerType, &internalConsumerID, tenantAccessLevels)
+	t.Logf("Create certificate subject mapping with subject: %s, consumer type: %s and tenant access levels: %s", subject, consumerType, tenantAccessLevels)
+	csmCreate := fixtures.CreateCertificateSubjectMapping(t, ctx, certSecuredGraphQLClient, csmInput)
+	defer fixtures.CleanupCertificateSubjectMapping(t, ctx, certSecuredGraphQLClient, csmCreate.ID)
+
+	// Create second certificate subject mapping
+	csmInput2 := fixtures.FixCertificateSubjectMappingInput(updatedSubject, updatedConsumerType, &updatedInternalConsumerID, updatedTntAccessLevels)
+	t.Logf("Create certificate subject mapping with subject: %s, consumer type: %s and tenant access levels: %s", updatedSubject, updatedConsumerType, updatedTntAccessLevels)
+	csmCreate2 := fixtures.CreateCertificateSubjectMapping(t, ctx, certSecuredGraphQLClient, csmInput2)
+	defer fixtures.CleanupCertificateSubjectMapping(t, ctx, certSecuredGraphQLClient, csmCreate2.ID)
+
+	certSubjectMappings := graphql.CertificateSubjectMappingPage{}
+	t.Log("Query certificate subject mappings...")
+	err = testctx.Tc.RunOperationWithoutTenant(ctx, certSecuredGraphQLClient, queryCertSubjectMappingsWithPaginationReq, &certSubjectMappings)
+	require.NoError(t, err)
+	require.NotEmpty(t, certSubjectMappings)
+	require.Equal(t, currentCertSubjectMappings.TotalCount+2, certSubjectMappings.TotalCount)
```

Comment:
Listed subject mappings could be asserted also

Reply:
I've added such assertions and for now, it will work but when the number of certificate subject mappings increases we would have to adapt the test to have pagination to assert the new values. That's why I left only the total count check.